src/features/Faucet/helpers.tsx

```diff
@@ -71,6 +70,27 @@ export const makeTxConfig = (
     baseAsset,
     networkId: network.id
   };
+  if ('maxFeePerGas' in txResult && 'maxPriorityFeePerGas' in txResult) {
+    /* Type 2 tx */
+    return {
+      ...txConfig,
+      rawTransaction: {
+        ...txConfig.rawTransaction,
+        maxFeePerGas: txResult.maxFeePerGas,
+        maxPriorityFeePerGas: txResult.maxPriorityFeePerGas,
+        type: 2
+      }
+    };
+  } else {
+    /* Type 1 tx */
+    return {
+      ...txConfig,
+      rawTransaction: {
+        ...txConfig.rawTransaction,
+        gasPrice: txResult.gasPrice
+      }
+    };
+  }
 };
```

Comment:
Maybe use [`makeTxConfigFromTx`](https://github.com/MyCryptoHQ/MyCrypto/blob/master/src/helpers/transaction.ts#L253).

Reply:
Agreed. @MrLuit If the responses from the Faucet API are all transaction params as hexlified strings (except chainId and type) they match type `ITxObject`. I would then use `makeTxConfigFromTx`. 

Nevertheless you can maybe convert your existing object if it doesn't match `ITxObject` entirely and the pass to `makeTxConfigFromTx` - should simplify a lot.

Reply:
Unfortunately `makeTxConfigFromTx` only seems to work for outgoing transactions (sender must be in `accounts: StoreAccount[]`). I'd say another helper for this funtionality would be nice to have but not a high priority as long as it's only used in the faucet.

Reply:
I would just say use it anyway and rewrite the incorrect data afterwards? That should still simplify your code a lot! Right?